src/NuGet.Clients/NuGet.PackageManagement.UI/PackageLoadContext.cs

```diff
@@ -50,6 +53,80 @@ internal class PackageLoadContext
 
         public Task<PackageCollection> GetInstalledPackagesAsync() =>_installedPackagesTask;
 
+        // get lock file for projects
+        public async Task<List<LockFile>> getLockFiles(IEnumerable<NuGetProject> nuGetProjects)
+        {
+            var nuGetProjectList = nuGetProjects.ToList();
+            var buildIntegratedProjects = nuGetProjectList.OfType<BuildIntegratedNuGetProject>().ToList();
+
+            List<LockFile> lockFiles = new List<LockFile>();
+            var lockFileFormat = new LockFileFormat();
+
+            foreach (var project in buildIntegratedProjects)
+            {
+                var lockFilePath = await project.GetAssetsFilePathAsync();
+                if (File.Exists(lockFilePath))
+                {
+                    lockFiles = lockFiles.Append(lockFileFormat.Read(lockFilePath)).ToList();
+                }
+            }
+            return lockFiles;
+        }
+
+        // get the dependent packages from the lock files for projects
+        public async Task<PackageCollection> GetDependentPackagesAsync()
+        {
+            // get lock files from projects. This will add all dependency packages for all projects in the list
+            List<LockFile> lockFiles = await getLockFiles(Projects.ToList());
+            List<PackageCollectionItem> dependentPackages = new List<PackageCollectionItem>();
+
+            foreach (LockFile lockFile in lockFiles)
+            {
+                foreach (LockFileTarget target in lockFile.Targets)
+                {
+                    foreach(LockFileTargetLibrary lib in target.Libraries)
+                    {
+                        foreach (Packaging.Core.PackageDependency dep in lib.Dependencies)
+                        {
+                            Versioning.NuGetVersion version;
+                            if(dep.VersionRange.HasUpperBound)
+                            {
+                                version = dep.VersionRange.MaxVersion;
+                            }
+                            else
+                            {
+                                version = dep.VersionRange.MinVersion;
+                            }
+                            PackageCollectionItem newItem = new PackageCollectionItem(dep.Id, dep.VersionRange.MaxVersion, null);
+                            if (!dependentPackages.Contains(newItem))
+                            {
+                                dependentPackages = dependentPackages.Append(new PackageCollectionItem(dep.Id, dep.VersionRange.MaxVersion, null)).ToList();
+                            }
+                        }
+                    }
+                }
+            }
+            return new PackageCollection(dependentPackages.ToArray());
+        }
+
+        // get target frameworks for BuildIntegratedNuGetProjects
+        public async Task<IEnumerable<string>> GetTargetFrameworksAsync()
```

Comment:
We have some code duplication here. 
The assets file contains the frameworks as well. If we have to read the assets file then we can avoid calling GetPackageSpecs. 

In general moving the abstraction to NuGet project probably works better. 

@rrelyea We discussed this in the past, but we should consider having a NuGet engineering add this abstraction or at the least work with @birgithi to help her navigate all the craziness we have.

Reply:
I re;moved this method here and added one to NuGetPackageManager.cs instead to get target frameworks. Let me know if there's a better way to do this.